includes/MediaWiki/Api/EditEndpoint.php

```diff
@@ -55,19 +57,32 @@ public function run() {
 		// TODO output an object that controls the reconciliations spec?
 		$inputReconcile = $request->reconcile();
 		if ( $inputReconcile === null ) {
-			die( 'Invalid reconcile JSON supplied' );
+			throw new LocalizedHttpException(
+				MessageValue::new( 'wikibasereconcileedit-editendpoint-invalid-reconcile-json' ),
+				400
+			);
 		}
+		$supportedReconciliationVersions = [ '0.0.1' ];
 		if (
 			!array_key_exists( self::VERSION_KEY, $inputReconcile ) ||
-			$inputReconcile[self::VERSION_KEY] !== '0.0.1'
+			!in_array( $inputReconcile[self::VERSION_KEY], $supportedReconciliationVersions )
 		) {
-			die( 'Only supported reconciliation version is 0.0.1' );
+			throw new LocalizedHttpException(
+				MessageValue::new( 'wikibasereconcileedit-editendpoint-unsupported-reconcile-version' )
+					->textListParams( $supportedReconciliationVersions )
+					->numParams( count( $supportedReconciliationVersions ) ),
+				400
+			);
 		}
 		if (
 			!array_key_exists( 'urlReconcile', $inputReconcile ) ||
 			!preg_match( PropertyId::PATTERN, $inputReconcile['urlReconcile'] )
 		) {
-			die( '0.0.1 requires a single urlReconcile key mapped to a property id, such as P123' );
+			throw new LocalizedHttpException(
+				MessageValue::new( 'wikibasereconcileedit-editendpoint-invalid-reconcile-propertyid' )
+					->textParams( '0.0.1', 'urlReconcile' ),
```

Comment:
Could we use $inputReconcile[self::VERSION_KEY] here already as we have validated it?

Reply:
Sure, makes sense.